src/Native/Unix/System.Net.Http.Native/pal_easy.cpp

```diff
@@ -214,14 +217,15 @@ static int seek_callback(void* userp, curl_off_t offset, int origin)
 extern "C" void
 HttpNative_RegisterSeekCallback(CURL* curl, SeekCallback callback, void* userPointer, CallbackHandle** callbackHandle)
 {
-    EnsureCallbackHandle(callbackHandle);
-
-    CallbackHandle* handle = *callbackHandle;
-    handle->seekCallback = callback;
-    handle->seekUserPointer = userPointer;
+    if (EnsureCallbackHandle(callbackHandle))
```

Comment:
It does, via the handle coming out as null. That's why I added the checks to the .cs file.